### DISCUSSION TITLE
Need help with @defer implementation details

### DISCUSSION THREAD

**Discussion by vamshiaruru-virgodesigns** [apollographql/router#3278] [Q&A]
Hi everyone, I want to use @defer in my production environments and I want to understand how it works before using it. I tried looking it up online, but I only find implementation details from the client side. What I want to understand is, how it is implemented on the apollo router itself.

Does it use server sent events? If yes, do I need to worry about the connection limit on SSEs? (For example you can only have 6 connections from same domain IIRC) Do I have to enable sticky sessions in my load balancer (I have two instances of apollo router running behind a load balancer)?

if it doesn't use SSE, does it use web sockets? In that case do I have to worry about sticky sessions again?

Any help is appreciated. If you could point to where I can read more about how apollo router implemented defer support, I'd love to read more on that. Thank you!

**Comment by Geal** [comment]
the `@defer` implementation uses a protocol similar to SSE: [multipart responses over chunked encoding](https://github.com/graphql/graphql-over-http/blob/main/rfcs/IncrementalDelivery.md). So it will hold an entire connection if requested over HTTP 1, but only for the duration of the response. After all the parts have been sent the connection can be used again to send a new request.

**Reply by vamshiaruru-virgodesigns** [reply]
Thank you, I'll read more on this :)
